File: level2_term2/DSA2/offline06/java/Main.java
package offline06.java;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.Scanner;

class Main {
    public static void main(String[] args) throws FileNotFoundException {
        String filename = "input.txt";
        File file = new File(filename);
        Scanner scanner = new Scanner(file);
        String command;
        MaxHeap max_heap = new MaxHeap();
        while (scanner.hasNext()) {
            command = scanner.next();
            if (command.equals("INC")) {
                int prev_value = scanner.nextInt();
                int new_value = scanner.nextInt();
                max_heap.increaseKey(prev_value, new_value);
            } else if (command.equals("INS")) {
                int value = scanner.nextInt();
                max_heap.insert(value);
                // max_heap.print();
            } else if (command.equals("FIN")) {
                int max = max_heap.findMax();
                System.out.println("FindMax returned " + max);
            } else if (command.equals("PRI")) {
                max_heap.print();
            } else if (command.equals("EXT")) {
                int max = max_heap.extractMax();
                // max_heap.print();
                System.out.println("ExtractMax returned " + max);
            }
        }
        scanner.close();
    }
}

File: level2_term2/DSA2/offline06/cpp/main.cpp
#include <iostream>
#include <fstream>
#include "max_heap.h"

using namespace std;

int main(int argc, char const *argv[])
{
    ifstream file;
    file.open("input.txt");
    MaxHeap max_heap;
    string command;
    while (!file.eof())
    {
        file >> command;
        if (command == "INC")
        {
            int prev_node;
            int new_node;
            cin >> prev_node >> new_node;
            max_heap.increase_Key(prev_node, new_node);
        }
        else if (command == "INS")
        {
            int node;
            file >> node;
            max_heap.insert(node);
            // max_heap.print();
        }
        else if (command == "FIN")
        {
            int max = max_heap.findMax();
            cout << "FindMax returned " << max << endl;
        }
        else if (command == "PRI")
        {
            max_heap.print();
        }
        else if (command == "EXT")
        {
            int max = max_heap.extractMax();
            cout << "FindMax returned " << max << endl;
        }
    }

    return 0;
}
